cli/internal/runsummary/run_summary.go

```diff
@@ -91,67 +140,86 @@ func (summary *RunSummary) Save(dir turbopath.AbsoluteSystemPath, singlePackage
 	return summaryPath.WriteFile(json, 0644)
 }
 
-// TaskSummary contains information about the task that was about to run
-// TODO(mehulkar): `Outputs` and `ExcludedOutputs` are slightly redundant
-// as the information is also available in ResolvedTaskDefinition. We could remove them
-// and favor a version of Outputs that is the fully expanded list of files.
-type TaskSummary struct {
-	TaskID                 string                                `json:"taskId"`
-	Task                   string                                `json:"task"`
-	Package                string                                `json:"package"`
-	Hash                   string                                `json:"hash"`
-	CacheState             cache.ItemStatus                      `json:"cacheState"`
-	Command                string                                `json:"command"`
-	CommandArguments       []string                              `json:"commandArguments"`
-	Outputs                []string                              `json:"outputs"`
-	ExcludedOutputs        []string                              `json:"excludedOutputs"`
-	LogFile                string                                `json:"logFile"`
-	Dir                    string                                `json:"directory"`
-	Dependencies           []string                              `json:"dependencies"`
-	Dependents             []string                              `json:"dependents"`
-	ResolvedTaskDefinition *fs.TaskDefinition                    `json:"resolvedTaskDefinition"`
-	ExpandedInputs         map[turbopath.AnchoredUnixPath]string `json:"expandedInputs"`
-	ExpandedOutputs        []turbopath.AnchoredSystemPath        `json:"expandedOutputs"`
-	Framework              string                                `json:"framework"`
-	EnvVars                TaskEnvVarSummary                     `json:"environmentVariables"`
-	Execution              *TaskExecutionSummary                 `json:"execution,omitempty"` // omit when it's not set
-	ExternalDepsHash       string                                `json:"hashOfExternalDependencies"`
-}
+// record sends the summary to the API
+// TODO: make this work for single package tasks
+func (rsm *Meta) record() []error {
+	errs := []error{}
+
+	// Right now we'll send the POST to create the Run and the subsequent task payloads
+	// when everything after all execution is done, but in the future, this first POST request
+	// can happen when the Run actually starts, so we can send updates to Vercel as the tasks progress.
```

Comment:
For context, here is the docs for bazel's build event protocol which streams updates (https://bazel.build/remote/bep)